### PR TITLE
DOC: Updated pointer spacing for consistency.

### PR DESCRIPTION
**Pull request by madphysicist**
I expect that this change will be pretty non-controversial. It was bugging me that the space came before the star in some cases and after in others. Then I went through the tutorial for something I needed, and ended up fixing some warnings and code formatting issued.

Ended up fixing what looks like a copy-and-paste bug in the tutorial code.

With commit e2516f1, this PR depends on #20689

### CODE REVIEW

**Comment by Mukulikaa** [comment]
Thanks @madphysicist, you seem to have got them all. I am not sure if these should have spaces as well:
https://github.com/numpy/numpy/blame/main/doc/source/user/c-info.ufunc-tutorial.rst#L875-L877

**Comment by madphysicist** [comment]
@Mukulikaa. I made some additional changes, including the ones you suggest. Thanks for the review. Please let me know if I missed anything.

File: doc/source/user/c-info.ufunc-tutorial.rst [hunk 1]
```diff
@@ -242,13 +239,14 @@ Example NumPy ufunc for one dtype
...
+For simplicity we give a ufunc for a single dtype, the ``'f8'``
+``double``. As in the previous section, we first give the ``.c`` file
+and then the ``setup.py`` file used to create the module containing the
+ufunc.
 
 The place in the code corresponding to the actual computations for
-the ufunc are marked with /\*BEGIN main ufunc computation\*/ and
-/\*END main ufunc computation\*/. The code in between those lines is
+the ufunc are marked with ``/\* BEGIN main ufunc computation \*/`` and
+``/\* END main ufunc computation \*/``. The code in between those lines is
```
Comment (mattip) [inline doc/source/user/c-info.ufunc-tutorial.rst:249]: Not for this PR, but at some point we should also point out that C++ templating can be used.

**Comment by mattip** [comment]
Thanks @madphysicist

**Comment by madphysicist** [comment]
@mattip technically, the example code is invalid until #20689 gets merged. The documentation says it's valid, but it will currently segfault everywhere the `data` argument to `PyUFunc_FromFuncAndData` is NULL.

**Pull request merged**